proposals/records.md

```diff
@@ -40,28 +40,37 @@ The synthesized members are as follows:
 
 Record types produce synthesized implementations for the following methods, where `T` is the
 containing type:
+```C#
+public override int GetHashCode();
+public override bool Equals(object other);
+public virtual bool Equals(T other);
+```
+The record type implicitly implements `System.IEquatable<T>` where `T` is the containing type.
 
-* `object.GetHashCode()` override
-* `object.Equals(object)` override
-* `T Equals(T)` method, where `T` is the current type
-* `Type EqualityContract` get-only property
-
-If either `object.GetHashCode()` or `object.Equals(object)` are sealed, an error is produced.
+Derived record types also override the `Equals(T other)` method from each base record type.
 
-`EqualityContract` is a virtual instance property which returns `typeof(T)`. If the base type
-defines an `EqualityContract` it is overridden in the derived record. If the base `EqualityContract`
-is sealed or non-virtual, an error is produced.
+The base record class synthesizes an `EqualityContract` property. The property is overridden in
+derived record classes. The synthesized implementations return `typeof(T)`.
+```C#
+protected virtual Type EqualityContract { get; }
+```
 
-`T Equals(T)` is specified to perform value equality such that `Equals` is true if and only if
-all accessible instance fields in the receiver are equal to the fields of the parameter
-and `this.EqualityContract` equals `other.EqualityContract`.
+If the base implementations of any of the overridden members is sealed or non-virtual,
+or does not match the expected signature and accessibility, an error is produced.
 
-`object.Equals` performs the equivalent of
+`Equals(T other)` performs value equality of the instance fields in this type with the fields of `other`.
+If there is a base record type, the field equality value is combined with `base.Equals(other)`.
+Otherwise the field equality value is combined with `EqualityContract.Equals(other.EqualityContract)`.
 
+The overrides of `Equals(T other)` for the base methods, including `object.Equals(object other)`, perform the equivalent of:
 ```C#
-override Equals(object o) => Equals(o as T);
+public override bool Equals(object other) => Equals(other as T);
 ```
 
+`GetHashCode()` generates a hash code by combining the values of `GetHashCode()` of the instance fields in this type.
+If there is a base record type, the field hash code is combined with `base.GetHashCode()`.
```

Comment:
>field hash code [](start = 36, length = 15)

"field hash code"?  Term not defined.

I recommend reorganizing as a bulleted list of things included in the hash code.